### PR TITLE
Redirect teachers to the page where they started sign up

### PR DESCRIPTION
**Pull request by pjanik**
- Redirects teachers to the page where they started sign up
- Adds `Portal.openSignupModal()` function, so Ethan can use it on landing pages
- Fixes some "Sign Up" and "regiter" links, they should open modal with sign up form now

Makes sense to review when https://github.com/concord-consortium/rigse/pull/241 is merged, as it's based on it.

### CODE REVIEW

**Comment by scytacki** [comment]
Can you rebase or merge master into this, so this PR only shows the new changes.

**Comment by pjanik** [comment]
@scytacki, done.

File: app/assets/javascripts/components/signup/signup.js.coffee [hunk 1]
```diff
@@ -35,6 +35,8 @@ modulejs.define 'components/signup/signup',
       anonymous: Portal.currentUser.isAnonymous
 
     onBasicDataSubmit: (data) ->
+      # Save the current path. Backed can use that information to redirect user to this page the first time he logs in.
```
Comment (scytacki) [inline app/assets/javascripts/components/signup/signup.js.coffee]: Do you mean Backend instead of Backed?

**Comment by scytacki** [comment]
👍  Looks good, just found one typo in a comment.

**Pull request merged**